valhalla/jawn/src/managers/IntegrationManager.ts

```diff
@@ -94,4 +94,76 @@ export class IntegrationManager extends BaseManager {
       }
     );
   }
+
+  public async getIntegrationByType(integrationType: string): Promise<
+    Result<
+      {
+        id: string;
+        integration_name: string;
+        settings: Record<string, any>;
+        active: boolean;
+      },
+      string
+    >
+  > {
+    const { data, error } = await supabaseServer.client
+      .from("integrations")
+      .select("id, integration_name, settings, active")
+      .eq("organization_id", this.authParams.organizationId)
+      .eq("integration_name", integrationType)
+      .single();
+
+    if (error) {
+      return err(error.message);
+    }
+
+    return ok(
+      data as {
+        id: string;
+        integration_name: string;
+        settings: Record<string, any>;
+        active: boolean;
+      }
+    );
+  }
+
+  public async getSlackChannels(): Promise<
```

Comment:
very solid method!

1. you check for org id
2. you wrapped Slack in a try catch